src/content/en/fundamentals/getting-started/primers/async-functions.md

```diff
@@ -0,0 +1,249 @@
+
+## Other async function syntax
+
+We've seen `async function() {}` already, but the `async` keyword can be used with other function syntax:
+
+### Arrow functions
+
+    // map some URLs to json-promises
+    const jsonPromises = urls.map(async url => {
+      const response = await fetch(url);
+      return response.json();
+    });
+
+Note: `array.map(func)` doesn't care that I gave it an async function, it just sees it as a function that returns a promise. It won't wait for the first function to complete before calling the second.
+
+### Object methods
+
+    const storage = {
+      async getAvatar(name) {
+        const cache = await caches.open('avatars');
+        return cache.match(`/avatars/${name}.jpg`);
+      }
+    };
+
+### Class methods
+
+    class Storage {
+      constructor() {
+        this.cachePromise = caches.open('avatars');
+      }
+
+      async getAvatar(name) {
+        const cache = await this.cachePromise;
+        return cache.match(`/avatars/${name}.jpg`);
+      }
+    }
+
+Note: Class constructors cannot be async.
+
+## Example: Streaming a response
+
+Note: If you're unfamiliar with streaming, [check out my guide](https://jakearchibald.com/2016/streams-ftw/#streams-the-fetch-api){: .external}.
+
+The benefit of async functions increases in more complex examples. Say we wanted to stream a response, logging out the chunks and returning the final size:
+
+Here it is with promises:
+
+    function getResponseSize(url) {
+      return fetch(url).then(response => {
+        const reader = response.body.getReader();
+        let total = 0;
+
+        return reader.read().then(function processResult(result) {
+          if (result.done) return total;
+
+          const value = result.value;
+          total += value.length;
+          console.log('Received chunk', value);
+
+          return reader.read().then(processResult);
+        })
+      });
+    }
+
+Look at that! See how I'm calling `processResult` inside itself to set up an asynchronous loop? Writing that made me feel *very smart*. But like most "smart" code you have to stare at it for ages to figure out what it's doing, like one of those magic-eye pictures from the 90's.
+
+Let's try that again with async functions:
+
+    async function getResponseSize(url) {
+      const response = await fetch(url);
+      const reader = response.body.getReader();
+      let result;
+      let total = 0;
+
+      while ((result = await reader.read()) && !result.done) {
+        const value = result.value;
+        total += value.length;
+        console.log('Received chunk', value);
+      }
+
+      return total;
+    }
+
+The asynchronous loop that made me feel so smug is replaced with a trusty, boring, while-loop. Much better.
+
+Note: The phrase "logging out the chunks" made me sick in my mouth.
+
+## Example: Avoid going too linear
```

Comment:
I didn't know what you meant by "too linear" when first reading this heading. It took me reading the examples to figure it out.

I wonder if sequential is a better word.